### DISCUSSION TITLE
Ingress messages were not routed to queues

### DISCUSSION THREAD

**Discussion by sandeepkr977** [rabbitmq/rabbitmq-server#11544] [Questions]
Hi Team, 

We are using RabbitMQ 3.13.2 in production. Last week we saw some unusual behaviour of Rabbit. Producers were able to produce messages successfully and we were able to see incoming messages flow as well but messages count in [Ready, Unacked, Total] showing 0 in Rabbit. Because of 0 message count consumers were not able consume messages. We had restarted many times consumers but still problem was same. When we have restarted the producer then problem had solved.

I have attached the screenshot for your reference.

<img width="1183" alt="Screenshot 2024-06-24 at 2 47 12 PM" src="https://github.com/rabbitmq/rabbitmq-server/assets/13862705/9aa11ea4-afe2-4b9e-a1ba-6b2b14dff695">
<img width="1431" alt="Screenshot 2024-06-19 at 3 13 45 PM" src="https://github.com/rabbitmq/rabbitmq-server/assets/13862705/47dff864-d016-4f71-aa4f-64c1cbbe349b">
<img width="1440" alt="Screenshot 2024-06-19 at 3 14 08 PM" src="https://github.com/rabbitmq/rabbitmq-server/assets/13862705/7c2cc23d-97d9-4cd1-854a-f4a190fb8f59">

**Comment by michaelklishin** [comment]
Your own screenshots provide evidence of messages being routed since about 15:05. We cannot know what was going on before that.

There are metrics for [returned and unroutable messages](https://www.rabbitmq.com/docs/publishers#unroutable). A messages that wasn't routed anywhere won't be persisted by definition.

**Comment by michaelklishin** [comment]
> When we have restarted the producer then problem had solved.

That means your topology is set up by the publishers on application boot. If something deletes bindings, for example, all published messages will go nowehere (and at best would be returned, if you publish them as mandatory).

Historically there has been one issue where bindings would disappear but no one could track it down. However, it involved transient queues and quorum queues are never transient. For 4.x with Khepri, we consider it to be solved because bindings are stored very differently.

A few screenshots cannot confirm or deny whether it was the case. It is certainly drastically less frequent than some 5-6 years ago.
